packages/components/src/components/rating-stars/rating-stars.tsx

```diff
@@ -24,30 +39,37 @@ export class RatingStars {
 
   @Element() hostElement: HTMLElement;
   /** (optional) hoverValue  */
-  @Prop({ mutable: true }) hoverValue = 0;
-  /** (optional) isHovering  */
-  @Prop({ mutable: true }) isHovering = false;
-  /** (optional) numOfStars  */
-  @Prop({ mutable: true }) numOfStars = 5;
-  /** (optional) rating  */
-  @Prop({ mutable: true }) rating = 0;
+  @State() hoverValue = 0;
+  /** (optional) hoverValue  */
+  @State() isHovering = false;
+  /** (optional) max  */
+  @Prop() max = 5;
+  /** (optional) value  */
+  @Prop({ mutable: true }) value = 0;
   /** (optional) small  */
-  @Prop({ mutable: true }) small = false;
+  @Prop() small = false;
   /** (optional) disabled  */
-  @Prop({ mutable: true }) disabled = false;
-  /** (optional) ariaTranslation  */
+  @Prop() disabled = false;
+  /** (optional) ariaLabelTranslation  */
   @Prop({ mutable: true })
-  ariaTranslation = `${this.rating} out of ${this.numOfStars} stars`;
+  ariaLabelTranslation = `${this.value} out of ${this.max} stars`;
   /** (optional) precision  */
   @Prop() precision = 1;
   /** (optional) slider label */
   @Prop() label?: string;
+  /** Emitted when the value has changed. */
+  @Event() scaleChange!: EventEmitter<void>;
 
   colorFilled = `var(--scl-color-primary)`;
   colorBlank = `var(--scl-color-grey-50)`;
   size = this.small ? '16px' : '24px';
 
-  getSymbol = (color: string, size: string, selected?: boolean) => {
+  @Watch('value')
+  handleValueChange() {
+    this.scaleChange.emit();
```

Comment:
This is a very clean an elegant technique (watching for changes in `value`), but it has a drawback: it can create a feedback loop (we made that mistake in the past haha).

`set value -> trigger event -> listen to event and update value -> set value -> ...`

So it's a bit more verbose emitting events in many places, but events should only be emitted after direct user interaction (mouse, touch).

Reply:
I have dropped the watcher. The emitter is now called on user interaction